webapp/migrations/Version20231229094359.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DoctrineMigrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version20231229094359 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return 'Add potential first to solve to scorecache.';
+    }
+
+    public function up(Schema $schema): void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->addSql('ALTER TABLE scorecache ADD is_potential_first_to_solve TINYINT(1) DEFAULT 0 NOT NULL COMMENT \'Is this potentially the first solution to this problem?\' AFTER is_first_to_solve');
```

Comment:
This seems to require a `_public` and `_jury` version: if you first solve a problem in the freeze and there's another one pending before yours, but also in the freeze, then the public/team should not know whether it was actually a first to solve or not.

In general, I'm a bit wary of adding extra complexity and columns. I think it would be better if we can figure out what scoreboard data depends what changes, and just fetch all that data and pass it to a single function that just recalculates the relevant scoreboard data.

Reply:
But we also don’t have a jury and public version of the normal first to solve. So I’m wondering if we have some logic, where we don’t show it if there are blue boxes anyway?

I get what you mean but I’m not sure how to solve this easily. A new incorrect submission can affect a correct submission that came after it and currently we only handle the current submission, not any other